File: 2714-LeftAndRightSumDifferences/2714-LeftAndRightSumDifferences.java
// Last updated: 31/07/2025, 21:07:18
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int[] leftSum=new int[nums.length];
        int[] rightSum=new int[nums.length];
        int[] answer=new int[nums.length];
        int sum=0;
        for(int i=-1;i<nums.length-1;i++)
        {
            if(i==-1)
            {
                sum+=0;
            }
            else
            {
                sum+=nums[i];
            }
            leftSum[i+1]=sum;
        }
        sum=0;
        for(int i=nums.length;i>=1;i--)
        {
            if(i==nums.length)
            {
                sum+=0;
            }
            else{
                sum+=nums[i];
            }
            rightSum[i-1]=sum;
        }
        for(int i=0;i<nums.length;i++)
        {
            answer[i]=Math.abs(leftSum[i]-rightSum[i]);
        }
        return answer;
    }
}

File: 2714-LeftAndRightSumDifferences/2714-LeftAndRightSumDifferences.py
# Last updated: 31/07/2025, 21:07:22
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        n=len(nums)
        answer=[0]*n
        leftsum=[0]*n
        rightsum=[0]*n
        for i in range(1,n):
            leftsum[i]=leftsum[i-1]+nums[i-1]
        for j in range(n-2,-1,-1):
            rightsum[j]=rightsum[j+1] + nums[j+1]
        for k in range(n):
            answer[k]=abs(leftsum[k]-rightsum[k])
        return answer                    